gmake 4.4

### Version of Apptainer

```
$ apptainer --version
apptainer version 1.1.8-1.el9
```
### Expected behavior

`make -j2` in a def file on a distribution that uses gmake >=v4.4 should work.

### Actual behavior

On distributions with gmake >=v4.4 a parallel make command fails with the following error.

```
make: *** cannot open jobserver /tmp/GMfifo157: No such file or directory.  Stop.
```

### Steps to reproduce this behavior

I created a throw-away repo to allow people to replicate this issue easily.  Just do the following:

```
$ git clone https://github.com/GodloveD/gmake-4.4-test.git

$ cd gmake-4.4-test/

$ apptainer build make.sif make.def 
```
The repo contains the following toy files:

- `hellomake.c`

```c
#include <hellomake.h>

int main() {
  // call a function in another file
  myPrintHelloMake();

  return(0);
}
```
- `hellofunc.c`

```c
#include <stdio.h>
#include <hellomake.h>

void myPrintHelloMake(void) {

  printf("Hello makefiles!\n");

  return;
}
```
- `hellomake.h`

```c
/*
example include file
*/

void myPrintHelloMake(void);
```

- `Makefile`

```make
CC=gcc
CFLAGS=-I.
DEPS = hellomake.h
OBJ = hellomake.o hellofunc.o 

%.o: %.c $(DEPS)
	$(CC) -c -o $@ $< $(CFLAGS)

hellomake: $(OBJ)
	$(CC) -o $@ $^ $(CFLAGS)
```

And then there is the def file used to produce the error.

```
Bootstrap: docker
From: fedora:39

%files
    hellomake.c
    hellofunc.c
    hellomake.h
    Makefile

%post
    dnf -y install gcc
    echo "**********"
    gmake --version
    echo "**********"
    make -j2
```
**NOTE:** This does not fail if you build the container with `sudo`!  :confused: 

Is there a suid binary or something that make is trying to use to create a fifo in `/tmp`?  

### What OS/distro are you running
I've replicated on a few different systems but here are details for one. 

```
$ cat /etc/os-release 
NAME="Rocky Linux"
VERSION="9.2 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.2"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.2 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.2"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.2"
```

### How did you install Apptainer

Unprivileged helper script.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,8 @@ For older changes see the [archived Singularity change log](https://github.com/a
 
 - Upgrade gocryptfs to version 2.4.0, removing the need for fusermount from
   the fuse package.
+- Fix seccomp filters to allow mknod/mknodat syscalls to create pipe/socket
+  and character devices with device number 0 for fakeroot builds.
 
 ## v1.2.0-rc.1 - \[2023-06-07\]
 

--- a/internal/pkg/runtime/engine/fakeroot/engine_linux.go
+++ b/internal/pkg/runtime/engine/fakeroot/engine_linux.go
@@ -165,10 +165,70 @@ func (e *EngineOperations) CreateContainer(context.Context, int, net.Conn) error
 // set the return value to 0 for mknod and mknodat syscalls. It
 // allows build bootstrap like yum to work with fakeroot.
 func fakerootSeccompProfile() *specs.LinuxSeccomp {
+	// sets filters allowing to create file/pipe/socket
+	// and turns mknod/mknodat as no-op syscalls for block
+	// devices and character devices having a device number
+	// greater than 0
 	syscalls := []specs.LinuxSyscall{
 		{
-			Names:  []string{"mknod", "mknodat"},
+			Names:  []string{"mknod"},
 			Action: specs.ActErrno,
+			Args: []specs.LinuxSeccompArg{
+				{
+					Index:    1,
+					Value:    syscall.S_IFBLK,
+					ValueTwo: syscall.S_IFBLK,
+					Op:       specs.OpMaskedEqual,
+				},
+			},
+		},
+		{
+			Names:  []string{"mknod"},
+			Action: specs.ActErrno,
+			Args: []specs.LinuxSeccompArg{
+				{
+					Index:    1,
+					Value:    syscall.S_IFCHR,
+					ValueTwo: syscall.S_IFCHR,
+					Op:       specs.OpMaskedEqual,
+				},
+				{
+					Index:    2,
+					Value:    0,
+					ValueTwo: 0,
+					Op:       specs.OpNotEqual,
+				},
+			},
+		},
+		{
+			Names:  []string{"mknodat"},
+			Action: specs.ActErrno,
+			Args: []specs.LinuxSeccompArg{
+				{
+					Index:    2,
+					Value:    syscall.S_IFBLK,
+					ValueTwo: syscall.S_IFBLK,
+					Op:       specs.OpMaskedEqual,
+				},
+			},
+		},
+		{
+			Names:  []string{"mknodat"},
+			Action: specs.ActErrno,
+			Args: []specs.LinuxSeccompArg{
+				{
+					Index:    2,
+					Value:    syscall.S_IFCHR,
+					ValueTwo: syscall.S_IFCHR,
+					Op:       specs.OpMaskedEqual,
+				},
+				{
+					Index:    3,
+					Value:    0,
+					ValueTwo: 0,
+					Op:       specs.OpNotEqual,
+				},
+			},
 		},
 	}
 	return &specs.LinuxSeccomp{

